### PR TITLE
pfj-31 improve error handling on provisioning

### PR DESCRIPTION
**Pull request by vitorestevamia**
This PR started with the objetive of handling errors on the invitation exchange, but finished with a general improvement and small refactor of the provisioning function

### CODE REVIEW

**Review by AnselmoLacerda — APPROVED** [review]

**Review by elazzarin — APPROVED** [review]

File: pkg/agent_deploy/agent.go [hunk 1]
```diff
@@ -42,7 +43,7 @@ func InstantiateAgent(agent ComposeInfo, hostName, profile string) error {
 
 	// append the rest of the command
 	command += fmt.Sprintf("compose -f /tmp/janus/docker-compose.yml --profile %s -p janus-%s up -d --no-recreate", profile, profile)
-	fmt.Println(command)
+	logger.InfoLogger("Executing command %s", command)
```
Comment (elazzarin) [inline pkg/agent_deploy/agent.go:46]:
```suggestion
	logger.InfoLogger("Running the %s command", command)
```

File: pkg/mqtt_pub/mqtt_pub.go [hunk 2]
```diff
@@ -2,32 +2,35 @@ package mqtt_pub
...
-type BrokerData struct {
-	BrokerServerIp string
-	BrokerPassword string
+type BrokerCredentials struct {
+	Username string
+	Password string
 }
 
-func PublishMessage(brokerData BrokerData, brokerUsername, publicationTopic string, sensorData map[string]any) {
+func PublishMessage(brokerIp string, credentals BrokerCredentials, sensorData map[string]any) error {
```
Comment (elazzarin) [inline pkg/mqtt_pub/mqtt_pub.go:18]:
```suggestion
func PublishMessage(brokerIp string, credentials BrokerCredentials, sensorData map[string]any) error {
```

File: pkg/mqtt_pub/mqtt_pub.go [hunk 3]
```diff
@@ -2,32 +2,35 @@ package mqtt_pub
...
+	publicationTopic := fmt.Sprintf("%s/attrs", credentals.Username)
```
Comment (elazzarin) [inline pkg/mqtt_pub/mqtt_pub.go:19]:
```suggestion
	publicationTopic := fmt.Sprintf("%s/attrs", credentials.Username)
```

File: pkg/mqtt_pub/mqtt_pub.go [hunk 4]
```diff
@@ -2,32 +2,35 @@ package mqtt_pub
...
 	}
 
 	mqtt.DEBUG = log.New(os.Stdout, "", 0)
 	mqtt.ERROR = log.New(os.Stdout, "", 0)
-	opts := mqtt.NewClientOptions().AddBroker(brokerData.BrokerServerIp + ":1883").SetClientID("dojot")
+	opts := mqtt.NewClientOptions().AddBroker(brokerIp + ":1883").SetClientID("dojot")
 
 	// Set username
-	opts.SetUsername(brokerUsername)
+	opts.SetUsername(credentals.Username)
```
Comment (elazzarin) [inline pkg/mqtt_pub/mqtt_pub.go:31]:
```suggestion
	opts.SetUsername(credentials.Username)
```

File: pkg/mqtt_pub/mqtt_pub.go [hunk 5]
```diff
@@ -2,32 +2,35 @@ package mqtt_pub
...
 	// Set password
-	opts.SetPassword(brokerData.BrokerPassword)
+	opts.SetPassword(credentals.Password)
```
Comment (elazzarin) [inline pkg/mqtt_pub/mqtt_pub.go:33]:
```suggestion
	opts.SetPassword(credentials.Password)
```

**Review by elazzarin — CHANGES_REQUESTED** [review]

**Review by elazzarin — APPROVED** [review]
Excellent job

**Pull request merged**